examples/angular/src/features/hacker-news-story/hacker-news-story.provider.module.ts

```diff
@@ -0,0 +1,24 @@
+import { HackerNewsStoryProvider } from './hacker-news-story.provider';
+import { GetHackerNewsLatestAskStoriesInteractor } from './domain/interactors/get-hacker-news-latest-ask-stories.interactor';
+import { GetHackerNewsStoryInteractor } from './domain/interactors/get-hacker-news-story.interactor';
+import { HackerNewsStoryDefaultProvider } from "./hacker-news-story.provider";
+
+export const HACKER_NEWS_PROVIDERS = [
+  {
+    provide: HackerNewsStoryProvider,
+    deps: [],
+    useFactory: () => {
+      return new HackerNewsStoryDefaultProvider();
+    },
+  },
+  {
+    provide: GetHackerNewsLatestAskStoriesInteractor,
+    deps: [HackerNewsStoryProvider],
+    useFactory: (provider: HackerNewsStoryProvider) => provider.getHackerNewsLatestAskStories(),
+  },
+  {
+    provide: GetHackerNewsStoryInteractor,
+    deps: [HackerNewsStoryProvider],
+    useFactory: (provider: HackerNewsStoryProvider) => provider.getHackerNewsStory(),
+  },
+];
```

Comment:
Since we're already implicitly coupled to Angular, why we don't just create an actual Angular module? or at least add a correct type to `HACKER_NEWS_PROVIDERS: Route[]` so that we get type checking.

Before someone else mentions this shouldn't be here. For me it's OK if we leave this file inside `features/<feature>/<feature>.provider.module.ts`; it's going to be much more practical than moving it to the `app` folder. Also, as long as the Angular coupling is limited to this file I'm fine with it.

Reply:
What I mean with "implicitly coupled to Angular" is that this array follows the `Route[]` structure that Angular expects, so, even if we're not importing anything Angular related it is indeed coupled. That's why I'm proposing to go full in and use Angular types/imports inside this file.

If you really want to be decoupled, I suppose we could define our types and a mapper to convert to `Route[]`.

My vote is for coupling.

Reply:
The actual type of this array is `Provider[]` which is also coupled to angular, so indeed, not having the explicit import doesn't meen we're not coupled, so if we're all ok with that, I'll leave it as an Angular module.

Reply:
lol, sorry, yes `Provider[]`… I've done too much Libris router refactoring lately. 😅

Reply:
@lucianosantana what do you thing about this? I think it's the last controversial change. See: https://github.com/mobilejazz/harmony-typescript/blob/eecf7f3b73aae8608db76ed4d3993f55533b90f7/examples/angular/src/features/hacker-news-story/hacker-news-story.provider.module.ts

Reply:
Creating an Angular module or having it typed as provider both makes sense to me. What doesn't make sense is that this file lives in the feature folder. 

For me, the harmony/business logic part finishes with the providing of interactors [in this file](https://github.com/mobilejazz/harmony-typescript/blob/eecf7f3b73aae8608db76ed4d3993f55533b90f7/examples/angular/src/features/hacker-news-story/hacker-news-story.provider.ts). The way the app wants to consume those interactors (all being injected in a single module or DI split in different modules) is a decision to be made from the app's perspective. 

Specifically I'd move [hacker-news-story.provider.module.ts](https://github.com/mobilejazz/harmony-typescript/blob/eecf7f3b73aae8608db76ed4d3993f55533b90f7/examples/angular/src/features/hacker-news-story/hacker-news-story.provider.module.ts) to `/examples/angular/src/app/story`

Reply:
I agree in part with what you say; this file is Angular code, hence it should go to the `app` folder. That's the correct thing to do.

But, I don't think it's the practical thing. There might or not a relation between `src/app` and `features` folders; for example, as you suggest we are already moving it into the `story` folder which doesn't match with `HackerNewsStory` feature. What happens when different UI modules use that provider? Where should we put it?

Leaving it next to the pure TS provider is convenient and easy to understand (although, _incorrect_).

Another option maybe is having a `providers` folder, or something like that, where we can put these provider modules… at least is less confusing where to put the file when we might have several UI modules that use the same provider.

I'm leaned towards the convenience option; although I'm OK also if we find a good place for these inside `app`.

Reply:
As @lucianosantana well pointed out, this provider is coupled to Angular so features folder looks not to be the right place to place it, but as we discussed, the UI and Features don't match in terms of structure so the best approach would be having a providers folder.

The points agains doing this is that in terms of development we need to create a provider for every feature, so it's completely linked and having this splitted in two folders is so unconvenient. Until we agree on another approach, I'm leaving as it is.